File: leet_413_arithmetic_slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums)
    {
        int len = nums.size();
        if (len < 3)
            return 0 ;

        int i = -1, j, res = 0, dif;
        while (++i < len - 2)
        {
            dif = nums[i + 1] - nums[i];
            j = i + 1 ;
            while (++j < len)
            {
                if (nums[j] - nums[j - 1] ^ dif)
                    break ;
                else
                    ++res ;
            }
        }
        return res ;
    }
};

File: leet_413_arithmetic_slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        size = len(nums)
        if size < 3:
            return 0
        res = 0
        for i in range(size - 2):
            dif = nums[i + 1] - nums[i]
            for j in range(i + 2, size):
                if nums[j] - nums[j - 1] != dif:
                    break
                else:
                    res += 1
        return res 
